mobilitydb/src/general/temporal_index.c

```diff
@@ -60,7 +61,8 @@ Temporal_gist_compress(PG_FUNCTION_ARGS)
   {
     GISTENTRY *retval = palloc(sizeof(GISTENTRY));
     Span *s = palloc(sizeof(Span));
-    temporal_bbox_slice(entry->key, s);
+    Temporal *temp = temporal_slice(entry->key);
+    temporal_set_bbox(temp, s);
```

Comment:
I know that `Temporal_gist_compress` should only be called for types whose bounding box is a span, but I would still suggest to use `temporal_set_tstzspan` instead of `temporal_set_bbox` here to avoid potential bugs in the future.
The same holds for other similar coccurences.

Generally, `temporal_set_bbox` should only be called with a `bboxunion` as second parameter, for example when testing for overlap between the boxes of two temporal types.
If the dimension of the second parameter is known (span, tbox or stbox), the corresponding function should be called (`temporal_set_tstzspan`, `temporal_set_tbox`, `temporal_set_stbox`, with the last two having an assert to make sure it is only called on types that have these dimensions).